network/network.go

```diff
@@ -89,16 +91,24 @@ func (n *Network) Walk(visitor dag.Visitor) error {
 }
 
 // NewNetworkInstance creates a new Network engine instance.
-func NewNetworkInstance(config Config, keyResolver types.KeyResolver, privateKeyResolver crypto.KeyResolver, didDocumentResolver types.DocResolver) *Network {
-	result := &Network{
+func NewNetworkInstance(
+	config Config,
+	eventsConnectionPool events.ConnectionPool,
+	keyResolver types.KeyResolver,
+	privateKeyResolver crypto.KeyResolver,
+	decrypter crypto.Decrypter,
```

Comment:
passing the crypto module twice is a bit weird. But lets tackle that in a refactor PR laer